examples/iterable-objects/iterable_objects.bal

```diff
@@ -1,36 +1,90 @@
 import ballerina/io;
 
-// An object that is a subtype of `Iterator<int>`.
+// A class that is a subtype of `Iterator<int, ()>`.
 class ArrayIterator {
     private int[] integers = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34];
     private int cursor = -1;
 
     // `next` method which generates the sequence of values of type `int`.
-    public function next() returns record {|int value;|}? {
+    public isolated function next() returns record {| int value; |}? {
         self.cursor += 1;
-        if (self.cursor < self.integers.length()) {
-            record {|int value;|} nextVal = {value: self.integers[self.cursor]};
+        if self.cursor < self.integers.length() {
+            record {| int value; |} nextVal = 
+                {value: self.integers[self.cursor]};
             return nextVal;
+        } else if self.cursor > self.integers.length() {
+            panic error("Index out of range");
```

Comment:
Thinking out loud, wonder if we really need this in this example. Given that we have another example that returns error here?

Reply:
Yes. TMI for this BBE. Remove the panic